parser.go

```diff
@@ -315,7 +321,20 @@ func parseLine(line string, recipe *Recipe) error {
 }
 
 func (p *ParserV2) parseLine(line string, recipe *RecipeV2) error {
-	if strings.HasPrefix(line, commentsLinePrefix) {
+	line = strings.TrimRight(line, " ")
+
+	if line == "---" && !p.inFrontMatter {
+		p.inFrontMatter = true
+	} else if line == "---" && p.inFrontMatter {
+		p.inFrontMatter = false
+		y := strings.NewReader(p.frontMatter)
+		err := yaml.NewDecoder(y).Decode(recipe.Metadata)
+		if err != nil {
+			return fmt.Errorf("decoding yaml front matter: %w", err)
+		}
+	} else if p.inFrontMatter {
+		p.frontMatter = p.frontMatter + line + "\n"
```

Comment:
nitpick: [strings.Builder](https://pkg.go.dev/strings#Builder) may be a better fit here.